cmplibrary/src/main/java/com/sourcepoint/cmplibrary/consent/ClientEventManager.kt

```diff
@@ -88,22 +95,20 @@ private class ClientEventManagerImpl(
         if (campaignsToProcess <= 0) {
             campaignsToProcess = Int.MAX_VALUE
 
-            val spConsent: SPConsents? = getSPConsents().getOrNull()
-            val spConsentString = spConsent
-                ?.let {
-                    spClient.onSpFinished(it)
-                    (spClient as? UnitySpClient)?.onSpFinished(it.toJsonObject().toString())
-                    it.toJsonObject().toString()
-                }
-                ?: run {
-                    spClient.onError(Throwable("Something went wrong during the consent fetching process."))
-                    "{}"
-                }
-            logger.clientEvent(
-                event = "onSpFinish",
-                msg = "All campaigns have been processed.",
-                content = spConsentString
-            )
+            getSPConsents().getOrNull()?.let {
+                val consentJson = JsonConverter.converter.encodeToString(it)
+                spClient.onSpFinished(it)
+                (spClient as? UnitySpClient)?.onSpFinished(consentJson)
```

Comment:
And it would be hard to find, debug and etc in this case

Reply:
@Nevazhnovu good point. Do we have tests for it?

Reply:
I doubt it's possible since the tests have to be performed in conditions that are hard to meet (unity + android + real device setup). I am unsure if the test exit in kotlin side